file-agent-kafka-client/src/main/java/im/ligas/kafka/stream/FileDataExtractor.java

```diff
@@ -1,23 +1,72 @@
 package im.ligas.kafka.stream;
 
+import org.apache.commons.codec.digest.DigestUtils;
+import org.apache.commons.io.FilenameUtils;
+
 import java.io.File;
 import java.io.IOException;
 import java.nio.file.Files;
 import java.nio.file.attribute.BasicFileAttributes;
+import java.util.ArrayList;
+import java.util.HashMap;
 
 public class FileDataExtractor {
+
+
     public FileData getFileData(File file) {
         FileData fileData = new FileData();
         fileData.setAbsolutePath(file.getAbsolutePath());
+        fileData.setId(DigestUtils.sha1Hex(fileData.getAbsolutePath()));
         fileData.setFileName(file.getName());
 
+        fileData.setSource(System.getProperty("user.home")); //What is the source system we should set to? IPaddress?
+
+        fileData.setTempLocation("/tmp/intermediate-data"); //File is stored with file-id in this temp location.
+
+        //Note: We need to fix on how we get the target system and properties of target system.
+        //For now i have made the default configurations.
+
+        //Setting the target system configurations.
+        ArrayList<TargetSystem> targetSystemList= new ArrayList<>();
```

Comment:
target and processors are not configured at this stage keep only the metadata section here